pkg/stanza/entry/field.go

```diff
@@ -33,6 +33,11 @@ type Field struct {
 	FieldInterface
 }
 
+// RootableField is a Field that may refer directly to "attributes" or "resource"
+type RootableField struct {
```

Comment:
@djaglowski is `RootableField` essentially a wrapper for `Field` that gets used in `ParserConfig` to help determine if a subfield is necessary or not?  The primary business logic happening in `newField`'s if statements?

Reply:
Yes, that's correct. The motivation for wrapping `Field` is to 1) implement `FieldInterface` easily and consistently, while 2) allowing for slightly different unmarshal/validation logic based on the type.